woocommerce/woocommerce-ios
# REST API: Authenticate users without WPCom login

Part of: #8453


## Description

This PR adds an entry point to authentication without WPCom. Changes include:
- A new feature flag to remove WPCom login requirement for site credential login flow. This feature flag is disabled by default to avoid blocking other teams - we need to manually enable this when working with application password until all features and migration are completed. 
- A placeholder site ID is used when WPCom login is bypassed. Currently, we require site ID in every view controller, so replacing it takes a big effort. The site ID is needed only for Jetpack tunneled requests and will be ignored for REST requests - so it should be safe to use a placeholder one for now. I'm open to any suggestions for a better solution 😅.

You'll notice that all features are not available after reaching the home screen. We'll need to migrate endpoints to make them work.

More enhancements will be handled in subsequent PRs: fetching site info, checking for role eligibility, application password availability, and WooCommerce plugin availability.

## Testing instructions

- Enable the feature flag `applicationPasswordAuthenticationForSiteCredentialLogin` and build the app.
- Log out of the app if needed.
- On the prologue screen, select "Enter your site address". If this option is not available, [disable simplified login](https://github.com/woocommerce/woocommerce-ios/blob/8d471f969911067311412d72f58de6ae2c966c6a/WooCommerce/Classes/Authentication/AuthenticationManager.swift#L61) and build the app again.
- Enter your test store address and select "Sign in with site credentials".
- Enter correct credentials and select "Continue". Notice that you are then navigated to the home screen.

## Screenshots


https://user-images.githubusercontent.com/5533851/208887257-3eeca2f8-b0aa-40d5-81de-8a5f91cbb824.mp4



---
